### ISSUE TITLE
[$4000]  mWeb/safari - Menu - "Open app banner" at the top causes spacing issues at the bottom of the browser

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#8791]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Pre-condition: User has the Expensify app installed on their device
1. Access staging.new.expensify.com
2. Sign into a valid applause.expensifail.com account
3. Tap on "Profile" icon on the top right (At this point there should be a banner in the header)
4 Tap on the "Workspace" to open the navigation links

## Expected Result:
The user expects to navigate the menus without any issues

## Actual Result:
The spacing that the banner takes up at the top of the screen ends of being compensated at the bottom once the user enters the workspace menus, this causes a visual blank space

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Mobile Web/safari
- Web

**Version Number:** 1.1.57.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**Email or phone of affected tester (no customers):** n/a

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/165404844-17752fa6-922f-49b3-8607-3c882deb637d.mp4

![Image from iOS (14)](https://user-images.githubusercontent.com/93399543/165404936-c53a6701-cff1-4a1b-8d99-d14e15612649.png)

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by cead22** [comment]
I was able to reproduce on staging but not production, so left it as deploy blocker and added Demolition

**Comment by roryabraham** [comment]
Just guessing, but https://github.com/Expensify/App/pull/8634 seems like the most likely culprit, followed by https://github.com/Expensify/App/pull/8521

**Comment by marcaaron** [comment]
The banner is an iOS [Smart App Banner](https://developer.apple.com/documentation/webkit/promoting_apps_with_smart_app_banners). Not smart enough apparently :trollface: 

That link says it's opt-in, but not seeing where we enabled it...

What outcome are we looking for here exactly?

**Comment by marcaaron** [comment]
>I was able to reproduce on staging but not production, so left it as deploy blocker and added Demolition

I'm not able to repro on dev so I'm removing the `Demolition` label.

**Comment by marcaaron** [comment]
Does seem to only affect the workspace page. So thinking a change related to that is possibly at fault.

**Comment by francoisl** [comment]
Having a hard time with this one too, it doesn't help that the smart banner doesn't show up on the dev site. 

I tried to use the standalone react devtools, but for some reason can't get it to connect to Safari on the simulator, even after following [this](https://reactnative.dev/docs/debugging#safari-developer-tools) - has anyone managed to get that working before? 

<img width="1435" alt="image" src="https://user-images.githubusercontent.com/2229301/165642819-83936edf-f1d1-425e-b8e9-76841d9c0c3d.png">

**Comment by sketchydroide** [comment]
did you end up getting anywhere with this @francoisl?

**Comment by francoisl** [comment]
Nope sorry, sounds impossible to track from a specific `<div>` what component caused it to be rendered.

**Comment by tgolen** [comment]
I had an initial look into this, and I had a hard time trying to get to the bottom of this. I could not determine any real difference between the workspace page and how it's structured and the `SecuritySettingsPage.js` which has a nearly identical structure, but not this bug. The only real differences are:

1. The workspace settings page has the area with name and avatar. These are all wrapped inside of views properly, so I cannot suspect this is the cause.
2. The workspace settings page has the extra items in the header. This doesn't appear to structurally change the layout of the page, but you never know...
3. The workspace settings page has a `<ConfirmModal>` at the bottom. This is the biggest suspect I have, but it's impossible to test locally since it cannot be reproduced locally.

What we might consider is at least eliminating `2` or `3` by temporarily removing one of the items and CPing it to staging.

This might help us narrow down the problem at least.

**Comment by sketchydroide** [comment]
I would say one of them, I will create a PR for number 3, seems to me like the main candidate from my experience on iOS, not that it really makes a bridge to RN, but I use what I can 😄

**Comment by sketchydroide** [comment]
not closed

**Comment by sketchydroide** [comment]
also I noted something else, this also happen on Search for a chort period of time. I then resets the frame. but it never shows the banner directing to the app store.
![IMG_4194](https://user-images.githubusercontent.com/242568/167139248-3fbea5f1-af89-447a-b2ff-9826af7ad808.PNG)

**Comment by sketchydroide** [comment]
Another thing I noticed is that if the user goes to another tab, or goes to background and comes back again the frame is redrawn and the view looks good again.

**Comment by sketchydroide** [comment]
I was trying to see if there was a React Native that could have been responsible for the change but did not find anything.

**Comment by chiragsalian** [comment]
Took some time to understand the issue, figure out if there was some way to test and research about the smart popover and possible solutions.
I tried possible solution 2 mentioned by tim [here](https://github.com/Expensify/App/issues/8791#issuecomment-1118001901). This is the [PR](https://github.com/Expensify/App/pull/8906). It didn't work ([screenshots](https://github.com/Expensify/App/pull/8906#issuecomment-1121709366)). 
Then i reverted it and tried to change some styles WorkspaceInitialPage.js - [PR](https://github.com/Expensify/App/pull/8909). That didn't work either. 
Next i've got a PR to revert the style changes i made. I also looked at the changelog for all the [navigation packages we use](https://github.com/react-navigation/react-navigation/tree/main/packages). While i don't see anything obvious i decided to yolo and try to update to latest to see if it helps - [PR](https://github.com/Expensify/App/pull/8912) 😅 (mostly running out of ideas)

**Comment by chiragsalian** [comment]
Updating navigation packages didn't help either so i'm reverting it. 

On a side note, since this is only affecting ios safari and isn't game breaking i think we should remove deploy blocker label and instead investigate this as a daily. This is because we haven't deployed newDot in 2 weeks and it would be nice to have the checklist and QA moving forward steadily unless we have strong pushback. Let me know what you guys think.

**Comment by sketchydroide** [comment]
I agree @chiragsalian, we can ask what everyone thinks in #deployer, but I think at this moment having a release is more important, and while this glitch is anoying it's not even permanent, is you go to another tab, or make safary go to background and then foreground again it fixes itself.

**Comment by tgolen** [comment]
I agree that it is OK to remove it as a deploy blocker. Thanks for all your
investigation and trial-and-error. This is super annoying, but maybe
something that we kick out to contributors and see if increasing the price
would attract some experts that can figure it out.

**Comment by chiragsalian** [comment]
Cool thanks for the input guys, removing this as a deployer blocker for now. Also yup getting more eyes on this with the help of contributors would be pretty nice too.

**Comment by mvtglobally** [comment]
Team noticed that this is also repro on Web 

https://user-images.githubusercontent.com/43995119/167948241-1e0ba78d-5fcb-4685-a2d5-5b3ae529b9b9.mp4

![Captura de Tela 2022-05-11 às 6 03 10 PM](https://user-images.githubusercontent.com/43995119/167948244-ac46a757-603c-4eff-89a2-8d5bcd78f6c3.png)

**Comment by mvtglobally** [comment]
MacBook Pro 2012 / catalina 10.15.7 / Safari @tgolen

**Comment by tgolen** [comment]
Can anyone on this thread that might know... point me to the decision that was made to use this smart banner? I'd be interested in knowing why that decision was made.

My reasoning is kind of like this:
- Do we need the smart banner?
- The app works perfectly fine on mobile web, so why are we pushing people to install the app?
- The app and mobile web have identical feature sets, so again, why are we pushing people to install the app?
- Isn't it actually preferrable to use mobile web since you'll have the most up-to-date code?

**Comment by luacmartins** [comment]
Sorry Tim, I don't know the answer to those questions, but I'm just stopping by to say that this issue sounds very similar to [this other issue](https://github.com/Expensify/App/issues/8731#issuecomment-1125367728). They both seem to be caused by a resizing of the window (either the app banner or the nav bar resizing). Maybe the issue I just linked is easier to test and the solution could potentially fix both?

**Comment by jrstanley** [comment]
My research around #8731 (suggesting no action, specifically because it is related to offsets and the presentation of the virtual keyboard) has, I believe, yielded the result to [the web repo](https://github.com/Expensify/App/issues/8791#issuecomment-1124297736) in @mvtglobally's issue. I'm unable to get the app banner to display on a real device and so I am unable to repro that.

Would you consider taking this external for me to send a proposal?

**Comment by chiragsalian** [comment]
Looks like steph is OOO. Reassigning.

**Comment by Santhosh-Sellavel** [comment]
@mallenexpensify Why was this HOLD for payment?

**Comment by mallenexpensify** [comment]
d'oh, I updated 'hold for payment' based on [a PR that was linked to this issue](https://github.com/Expensify/App/pulls?q=is%3Apr+https%3A%2F%2Fgithub.com%2FExpensify%2FApp%2Fissues%2F8791+is%3Aclosed) but it looks like it was incorrect so removed 'hold from payment' from the title and the 'awaiting payment' label.  :ohnothing:

**Comment by AndrewGable** [comment]
Double please!

**Comment by mallenexpensify** [comment]
Doubled price to $500
https://www.upwork.com/jobs/~01212954502386b607

**Comment by Santhosh-Sellavel** [comment]
I'm going on OOO. But I still keeping this assigned to me. If there are any immediate C+ duties needed here feel free to unassign me & re-apply exported label thanks!

**Comment by mallenexpensify** [comment]
Will do @Santhosh-Sellavel !

**Comment by liyamahendra** [comment]
Based on what @mvtglobally reported [here](https://github.com/Expensify/App/issues/8791#issuecomment-1124297736), I did some local testing for the fix, which I believe solves this issue.

But apparently, since the iOS Smart banner can't be shown while running the site locally, is there a provision to test my changes on staging for a brief moment and revert it immediately? This will allow me to verify my fix.

**Comment by AndrewGable** [comment]
Yes, we can try that.

**Comment by mallenexpensify** [comment]
Doubled price to $1000

**Comment by liyamahendra** [comment]
# Proposal

The Workspace screens are under the `SettingsModalStackNavigator` created using `createModalStackNavigator`. The proposal for the fix is to alter the `cardStyle` by making sure it expands to the full screen height width. 

The fix is to alter the `navigationScreenCardStyle` as below:

```
 navigationScreenCardStyle: {
        backgroundColor: themeColors.appBG,
        height: '100%',
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
    },
```

**Comment by Santhosh-Sellavel** [comment]
Will review this later today.

**Comment by Santhosh-Sellavel** [comment]
> Based on what @mvtglobally reported [here](https://github.com/Expensify/App/issues/8791#issuecomment-1124297736), I did some local testing for the fix, which I believe solves this issue.
> 
> But apparently, since the iOS Smart banner can't be shown while running the site locally, is there a provision to test my changes on staging for a brief moment and revert it immediately? This will allow me to verify my fix.

@liyamahendra 
We should look for the steps to reproduce the issue locally.

> I did some local testing for the fix, which I believe solves this issue.
Can share testing steps.

Please close your PR since your proposal is not accepted yet, thanks!

**Comment by liyamahendra** [comment]
> @liyamahendra
We should look for the steps to reproduce the issue locally.

The issue can’t be reproduced on local dev website because the iOS Smart Banner doesn’t show up for locslhost. I believe you’re aware about this?

What I meant to say with local testing is - running the staging website locally and applying the fix using developer tools. 

The PR was only created so that we could do a quick testing on staging and then revert back. This was in response to @AndrewGable saying that we can do a quick testing on staging for the fix and verify. https://github.com/Expensify/App/issues/8791#issuecomment-1153917051

I can definitely close the PR, just need few minutes to test out my fix and verify if it does solve the issue.

@Santhosh-Sellavel

**Comment by tgolen** [comment]
Since there were no answers to my questions here: https://github.com/Expensify/App/issues/8791#issuecomment-1125243282

As long as people agree with my logic, can we please just remove this banner?

**Comment by Santhosh-Sellavel** [comment]
> Since there were no answers to my questions here: [#8791 (comment)](https://github.com/Expensify/App/issues/8791#issuecomment-1125243282)
> 
> As long as people agree with my logic, can we please just remove this banner?

I'm not sure but smart banners are like part of the deep linking feature.

**Comment by tgolen** [comment]
Can you explain that more, please? 

- Why do we need a banner for deep links?
- Where are we using the deep links?
- Will they continue to work without deep links?

**Comment by Santhosh-Sellavel** [comment]
> Why do we need a banner for deep links?
Banner is used to help us show there is a native app installed available for use.

> Where are we using the deep links?

Check the below image this is our config for deeplink/applink. Taking out an example, consider user installs our app for the first time from the app store. Signs up using phone/email, now password setup link is sent. When the user taps the link from message/mail user is redirected back to our app as we listen to `setpassword` path.

<img width="752" alt="Screenshot 2022-06-17 at 11 41 05 PM" src="https://user-images.githubusercontent.com/85645967/174354846-caf66ba3-36f5-46e8-832a-8a65040860e7.png">
There is a typo in comment for setpassword path

Will they continue to work without deep links?

I'm not sure but think it might not be without deep links?

**Comment by tgolen** [comment]
> us show there is a native app installed available for use

OK, since it's not a requirement, and this is not something that we have a well-defined problem for, and I can argue against it (like I did in my original post), then it doesn't make any sense to keep it.

> Will they continue to work without deep links?

Ah, sorry, I mispoke here. What I meant was... "Will the deep links still work without the banner?"

**Comment by Santhosh-Sellavel** [comment]
> "Will the deep links still work without the banner?"

I'm not 100% sure. Because I was wondering about how the smart banner shows up as we don't have any implementation as said here -> https://developer.apple.com/documentation/webkit/promoting_apps_with_smart_app_banners

But based on this [thread](https://developer.apple.com/forums/thread/105129) and the below comment clears everything

<img width="1217" alt="Screenshot 2022-06-21 at 12 28 49 AM" src="https://user-images.githubusercontent.com/85645967/174663766-79ac197e-72ec-431b-91b2-331c97d65f5d.png">

As per hunch, they use the apple-app-site-associations to show without any implementation. 

Conclusion: In our app smart banner is shown only because of deep link integration.

**Comment by tgolen** [comment]
Ah, that's interesting! That explains why it shows on some pages and not
others. I wish we could disable that :(

**Comment by Santhosh-Sellavel** [comment]
Can we double this up or bump it down to monthly?

**Comment by mallenexpensify** [comment]
Thanks for the ping @Santhosh-Sellavel , I doubled now, since it seems like something we want to fix.  If anyone thinks it should be monthly, chime in.  
https://www.upwork.com/jobs/~01212954502386b607

**Comment by liyamahendra** [comment]
# Proposal

So, this was a really tricky one as it was tough to replicate locally. Following are the steps to reproduce it locally:

1. Add below entry to /etc/hosts (Mac)
`127.0.0.1	staging.new.expensify.com`
2. Configure server to run on port 80 and https by modifying `webpack.dev.js`:
```
return merge(baseConfig, {
        mode: 'development',
        devtool: 'inline-source-map',
        devServer: {
            contentBase: path.join(__dirname, '../../dist'),
            hot: true,
            ...proxySettings,
            historyApiFallback: true,
            disableHostCheck: true,
            https: true,
            port: 443,
        },
    });
```
3. Since OS only allows processes from privileged users to bind to port 1024 and below, run the web server with `sudo` as `sudo npm run web`
4. Since a self-signed certificate is used, trust the website and visit it.

### Solution

The Workspace screens are under the SettingsModalStackNavigator created using createModalStackNavigator. The style of these screens can be configured by the `cardStyle` which is currently set to:
```

```
navigationScreenCardStyle: {
        backgroundColor: themeColors.appBG,
        height: '100%',
},
```

Change this style as below:
```
navigationScreenCardStyle: {
    backgroundColor: themeColors.appBG,
    ...Platform.select({
        web: {
            height: '90vh'
        },
        default: {
            height: '100%',
        },
    }),
},
```

Tested this on both staging (where the smart banner is shown) and localhost (where no smart banner is shown) and on the native platform.

### With smart banner visible

https://user-images.githubusercontent.com/911879/176687434-a1c6a96e-2b88-4041-9279-5185ab1723b6.mp4


### Without smart banner visible

https://user-images.githubusercontent.com/911879/176687490-64b1bab4-b625-4a65-8bf6-d62992472d66.mp4


### Native Platform

https://user-images.githubusercontent.com/911879/176687555-3f0297b8-c427-4b52-9259-43215d96c77b.mp4

@marcaaron @Santhosh-Sellavel kindly review and share your comments.

**Comment by mallenexpensify** [comment]
@Santhosh-Sellavel can you please review the above proposal? Reassigning a CM since I'll be OOO

**Comment by Santhosh-Sellavel** [comment]
Oops I forgot to leave a comment on the proposal. The proposal looks like a work around to me.


@liyamahendra Can you elaborate more on how the change addresses the issue?

**Comment by liyamahendra** [comment]
@Santhosh-Sellavel mind sharing why you feel the solution is a workaround?

As far as I understand my solution is a fix to the issue. The only workaround is to reproduce the issue locally.

iOS mobile safari browser has issues with *height: 100%* and many blogs you will find that the recommendation is to use *vh*. The changes I proposed ensures that the height expands properly taking care of the smart banner height from the previous display.

**Comment by liyamahendra** [comment]
> iOS mobile safari browser has issues with height: 100% and many blogs you will find that the recommendation is to use vh.

I realised that ours was a specific use case, so edited my comment, avoiding references to the recommendations I mentioned. 

> Also 100% height doesn't cause us any issue for our app, does it ?

Since ours is a specific use case, I'll share details in a separate comment as to why **height: 100%** doesn't work well here.

**Comment by liyamahendra** [comment]
This comment is in two parts:

1. For why **height: 100%** causes an issue

> Also 100% height doesn't cause us any issue for our app, does it ?

**height: 100%** works normally but fails in our particular use case due to Smart banner. 

To understand this properly, we need to clearly understand the difference between **height: 100%** and the use of **vh**.

- When **height: 100%;** is used, the element to which this attribute is applied, is sized 100% in height relative to the size of its parent element.
- When **height: 100vh** is used, the element to which this attribute is applied, will be 100% of the viewport height regardless of where the element resides in the DOM.

Now talking about our specific Usecase:

The Smart Banner is shown on the Settings Page. And, the Workspace pages are shown as Modal via createModalStackNavigator i.e. the Parent of the Workspace Modal pages will be the Settings page. 

We'll understand the usecase via numbers. Let's say the Settings Page height is 1000px and the height of smart banner is 100px. When the Smart bnnner is visible, the resulting height of the Settings page becomes (1000 - 100) = 900px.

Now when the Workspace pages Modal is shown, the OS doesn't show the Smart banner. The height of this modal pages is set to **height: 100%** so it will try to extend the height of modal to 100% height of the parent which is Settings page, whose height had been reduced by 100px. So, the Workspace Modal height becomes 900px and shows the 100px overlay background. Doesn't that make sense?

2. Why I used 90vh instead of 100vh

Have a read through here: https://www.lullabot.com/articles/unexpected-power-of-viewport-units-in-css

<img width="1074" alt="image" src="https://user-images.githubusercontent.com/911879/177789943-1b3cfc0b-ac3b-4e1a-9f4a-33f5f64b55be.png">

**90vh** is used to ensure that the Modal element's height remains shorter than the viewport.

**Comment by Santhosh-Sellavel** [comment]
> > iOS mobile safari browser has issues with _height: 100%_ and many blogs you will find that the recommendation is to use _vh_.
> 
> Okay `vh` is fine. Why is it `90 vh` it should be `100 vh` right? Also 100% height doesn't cause us any issue for our app, does it ?
> 
> Can you share those additional links for those recommendations let us have a look at it.

@liyamahendra 
As quoted above, you mentioned there were many blogs recommending it. I don't find any such thing.  Seems that's statement removed from the comment.

I'm not on board with your proposal. I find nothing wrong with CSS. If I access the workspace directly or reload on the workspace page everything looks fine.

I think either we should show a banner on all pages or not show a banner at all. 

cc: @AndrewGable @tgolen

**Comment by liyamahendra** [comment]
> If I access the workspace directly or reload on the workspace page everything looks fine.

Can you share a recording of this?

**Comment by Santhosh-Sellavel** [comment]
https://user-images.githubusercontent.com/85645967/178314744-38a9edd1-b5dc-4c37-9bfa-2f2a5811182c.MOV

**Comment by Santhosh-Sellavel** [comment]
I am navigating to the workspace page and refreshing the page in the video.

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad You can try out these steps mentioned in the https://github.com/Expensify/App/issues/8791#issuecomment-1171213402.

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel I tried to test with the above-mentioned steps but it does not work. I also tried to remove some of the existing links but that does not remove the same banner from the web app. I think we have to test it on the staging server.

**Comment by Santhosh-Sellavel** [comment]
@AndrewGable 
We have not implemented the smart banner, so why not try implementing it. 

So here is my proposal
I've experimented with it locally by following the steps mentioned [here](https://developer.apple.com/documentation/webkit/promoting_apps_with_smart_app_banners)

Just need to add the below line in [index.html](https://github.com/Expensify/App/blob/main/web/index.html) under `head`
```html
<meta name="apple-itunes-app" content="app-id=1530278510, app-argument='/'">
```
Here is the behavior of the implementation.
https://user-images.githubusercontent.com/85645967/178562184-ec4218ce-163a-40f3-8346-7ffa2b126a99.mov

Now there is an option to dismiss the banner, after dismissing the banner it never showed up until I cleared the cache data.

Note we need to test this on staging to ensure whether this overrides the deep link config i.e which cause this issue in the first place.

cc: @tgolen

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel Does it show the app banner on workspace page or does it fixes the issue when you did not dismiss the banner?

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad Uploaded the wrong video. Updated the video now.

> Does it show the app banner on the workspace page

Yes, it shows the app banner on the workspace page also.

> Does it fixes the issue when you did not dismiss the banner?

Yes, it fixes.

I'm unsure about one thing, currently, we don't have smart banner implementation. The banner shown now is based on the deep link config from apple-site-association-file. Only after testing on the staging, we can confirm whether this overrides the existing banner or not.

cc: @AndrewGable @tgolen

**Comment by stephanieelliott** [comment]
Hey @JmillsExpensify, reassigning this one since I am about to head OOO. We're still reviewing proposals on this one!

**Comment by Santhosh-Sellavel** [comment]
Any thoughts @AndrewGable?

**Comment by Santhosh-Sellavel** [comment]
Waiting for thoughts @AndrewGable or @tgolen

**Comment by tgolen** [comment]
Let's give the smart banner a try! What have we got to lose? Let's hire @Santhosh-Sellavel for this.

**Comment by JmillsExpensify** [comment]
Invited @Santhosh-Sellavel to the job here: https://www.upwork.com/jobs/~01212954502386b607

**Comment by Santhosh-Sellavel** [comment]
Accepted invite, let's remove the `Help Wanted` label here.

cc: @AndrewGable @JmillsExpensify

**Comment by JmillsExpensify** [comment]
Hired @Santhosh-Sellavel on Upwork. Thanks!

**Comment by JmillsExpensify** [comment]
Reviewers have requested updates in the linked PR. We're currently working through those.

**Comment by Santhosh-Sellavel** [comment]
So we experimented with the smart banner and got to know deep link banner overrides it too. 

We know as I said [early](https://github.com/Expensify/App/issues/8791#issuecomment-1160760539) banner is shown because of paths configuration in `apple-site-association-file`. 

Now the options are,

1. To show this banner across other pages, we should update the `apple-site-association-file` with paths that are not configured yet for deep linking.
2. Look for any alternate solutions mostly it would be workarounds
3. Do nothing for now revisit it later, as this deep link banner would show only if the app is already installed, as most users won't prefer using mWeb while the native app is already installed.

cc: @AndrewGable @tgolen

**Comment by tgolen** [comment]
Yeah, I'm mostly in the camp of "do nothing" for now. At least we
understand the behavior better, but it's not really a problem. Just
kinda... sucks.

**Comment by mdneyazahmad** [comment]
@Santhosh-Sellavel 

> We know as I said https://github.com/Expensify/App/issues/8791#issuecomment-1160760539 banner is shown because of paths configuration in apple-site-association-file.

 I think this change was initially proposed by me [here](https://github.com/Expensify/App/issues/8791?notification_referrer_id=NT_kwDOBKKL07MzNTE3NTM3NjcwOjc3NzYxNDkx&notifications_query=reason%3Aparticipating#issuecomment-1181993100). We should fix this bug not because of the smart banner but it creates empty space at the bottom.

**Comment by JmillsExpensify** [comment]
>Do nothing for now revisit it later, as this deep link banner would show only if the app is already installed, as most users won't prefer using mWeb while the native app is already installed.

I'm in this camp as well.

**Comment by AndrewGable** [comment]
Ok doing nothing, so let's close. Reopen if you disagree.

**Comment by Santhosh-Sellavel** [comment]
@AndrewGable 

Does this job qualifies for payment, as proposed solution was implemented but the result was not as we expected though?

cc: @JmillsExpensify @tgolen

**Comment by mdneyazahmad** [comment]
I have been following this issue for a long time and also proposed a solution. In general higher prices issue is not that easy and it takes a quite amount of time to find the root cause and propose a working solution.

I proposed https://github.com/Expensify/App/issues/8791#issuecomment-1181993100 a solution that would have been fixed the issue but it is not that easy to test the solution locally. We had multiple different proposals in past and had created multiple PRs to test the solution and that did not work so reverted. 

@Santhosh-Sellavel  has suggested https://github.com/Expensify/App/issues/8791#issuecomment-1182008045 to test my proposal and meanwhile he also had a proposal that he tested locally and thought that it will fix the issue. He has been assigned. This PR also does not work as expected and he came know that the same solution as proposed by me will fix the issue and he also suggested to close and the issue is closed.

@Santhosh-Sellavel your original https://github.com/Expensify/App/issues/8791#issuecomment-1160760539 says 
> "Conclusion: In our app smart banner is shown only because of deep link integration."

I think this is not not enough to fix the issue, it just states the banner is there because of the deep link integration and does not explain why there is a gap on the some pages and what are the exact code changes that will be needed to show and not show banner on some changes.

I would appreciate it, if I also get compensated for the time as I proposed a solution before it was closed and my proposal was never tested.

cc: @JmillsExpensify @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
@mdneyazahmad 

> @Santhosh-Sellavel your original https://github.com/Expensify/App/issues/8791#issuecomment-1160760539 says
> 
> "Conclusion: In our app smart banner is shown only because of deep link integration."
> 
> I think this is not not enough to fix the issue, it just states the banner is there because of the deep link integration and does not explain why there is a gap on the some pages and what are the exact code changes that will be needed to show and not show banner on some changes.

In the same [comment]( https://github.com/Expensify/App/issues/8791#issuecomment-1160760539) I have mentioned 

> they use the apple-app-site-associations to show without any implementation. 

That's enough to show why the app banner shows on some pages (deep link configured) alone. If not, a couple
 of comments earlier I posted the entire configuration while answering the questions @tgolen [here](https://github.com/Expensify/App/issues/8791#issuecomment-1159129647)

We know the root cause of why the banner is shown as detailed by this comment on Jun 21 - https://github.com/Expensify/App/issues/8791#issuecomment-1160760539. So it's clear at this point modifying the deep link config alone is enough to show/hide the app banner, we are still looking for other options.

Indeed your proposal that came on Jul 12 https://github.com/Expensify/App/issues/8791#issuecomment-1181993100 was to add workspace path to the config file

I acknowledged the same in https://github.com/Expensify/App/issues/8791#issuecomment-1182008045 and my comment was
> Also, we have a new [proposal](https://github.com/Expensify/App/issues/8791#issuecomment-1181993100) from @mdneyazahmad this would show "App banner" on the workspace page also.

Meanwhile, I found the smart banner solution which I proposed here https://github.com/Expensify/App/issues/8791#issuecomment-1182160986, got hired, and the rest is clear.

cc: @AndrewGable @tgolen @JmillsExpensify

**Comment by tgolen** [comment]
We are discussing this internally to see how to best move forward.

**Comment by mallenexpensify** [comment]
Sorry this took forever.  There was a lot of discussion . 

@Santhosh-Sellavel was hired to fix the job.  Due to limitations beyond the control of us/Expensify and Santhosh, the bug hasn't been fixed yet.  As time has passed working on this issue and we've learned more about the issue and options to fix it, we've decided to close this issue and not move forward with a fix.  If @Santhosh-Sellavel would have completely fixed the issue and the bug didn't still persist, we'd compensate the full amount.  Since the bug currently persists, we have decided to compensate $2000 for the edge case.  

@mdneyazahmad , unfortunately, you're ineligible for compensation for this issue.  At a high level, time invested before an accepted proposal isn't due compensation (Time after neither cuz we've already hired someone who we'll be paying)  

The end result of @Santhosh-Sellavel fixing or not fixing the issue doesn't factor into the fact that their proposal was accepted and that they were hired. 

We appreciate the time you invested, if you think our stance here with this issue isn't aligned with our [CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md), please comment on this issue and reference the CONTRIBUTING.md where applicable.  

@Santhosh-Sellavel has been paid $2000 in Upwork and their contract ended

**Comment by JmillsExpensify** [comment]
I recently reported [this bug](https://expensify.slack.com/archives/C049HHMV9SM/p1668119557057829), after I clearly forgot being part of the discussion originally. So it's still an odd experience to run into despite what we decided. That kind of got me to thinking, why do we have the "open app banner" at the top of the page? We're spending quite a lot of effort to make the app work on all platforms, so it struck me as a bit odd that we would keep pushing the native app on mWeb, really at the expense of the mWeb experience. Maybe we should just remove that banner to begin with.

**Comment by Santhosh-Sellavel** [comment]
~why do we have~  why the "open app banner" at the top of the page?  

Please read [this](https://github.com/Expensify/App/issues/8791#issuecomment-1160760539) 

Cc: @JmillsExpensify

**Comment by JmillsExpensify** [comment]
That doesn't answer my question. I'm asking why we have an app banner at all.

**Comment by Santhosh-Sellavel** [comment]
That's what I am trying to answer, we didn't implement banner. Banner was added automatically by iOS Safari browser based deeplink configuration.

**Comment by tgolen** [comment]
@JmillsExpensify Yeah, I had asked the exact same questions earlier. @Santhosh-Sellavel is right. This is not something that we control. Because of the deep-linking configuration that we have to upload our app with (to get deep-links to work), Apple will automatically display that banner on any page which can be deep-linked to. It sucks, and we can't control it :(

**Comment by JmillsExpensify** [comment]
Oh thanks both. I'm on the same page now, appreciate the more detailed explanations. Keeping this close in any case for now.
